backend/utils/secure_temp_files.py

```
@@ -0,0 +1,402 @@
+"""
+Secure temporary file handling utilities for StudyMate-v2.
+
+This module provides secure temporary file creation with proper cleanup,
+restrictive permissions, and optional encryption for sensitive data.
+"""
+
+import os
+import tempfile
+import subprocess
+from contextlib import contextmanager
+from pathlib import Path
```

Ruff (0.12.2)

`pathlib.Path` imported but unused

Remove unused import: `pathlib.Path`

(F401)